airbyte-bootloader/src/main/java/io/airbyte/bootloader/SecretMigrator.java

```diff
@@ -0,0 +1,206 @@
+              () -> UUID.randomUUID());
+          source.setConfiguration(migratedConfig);
+          return source;
+        })
+        .toList();
+
+    for (final SourceConnection source : sourceConnections) {
+      configPersistence.writeConfig(ConfigSchema.SOURCE_CONNECTION, source.getSourceId().toString(), source);
+    }
+  }
+
+  /**
+   * This is migrating the secrets for the destination actors
+   */
+  @VisibleForTesting
+  void migrateDestinations(final List<DestinationConnection> destinations, final Map<UUID, JsonNode> definitionIdToDestinationSpecs)
+      throws JsonValidationException, IOException {
+    log.info("Migration Destinations");
+
+    final List<DestinationConnection> destinationConnections = destinations.stream().map(destination -> {
+      final JsonNode migratedConfig = migrateConfiguration(new ConnectorConfiguration(
+          destination.getWorkspaceId(),
+          destination.getConfiguration(),
+          definitionIdToDestinationSpecs.get(destination.getDestinationDefinitionId())),
+          () -> UUID.randomUUID());
+      destination.setConfiguration(migratedConfig);
+      return destination;
+    })
+        .toList();
+    for (final DestinationConnection destination : destinationConnections) {
+      configPersistence.writeConfig(ConfigSchema.DESTINATION_CONNECTION, destination.getDestinationId().toString(), destination);
+    }
+  }
+
+  /**
+   * This is a generic method to migrate an actor configuration It will extract the secret path form
+   * the provided spec and then replace them by coordinates in the actor configuration
+   */
+  @VisibleForTesting
+  JsonNode migrateConfiguration(final ConnectorConfiguration connectorConfiguration, final Supplier<UUID> uuidProvider) {
+    if (connectorConfiguration.getSpec() == null) {
+      throw new IllegalStateException("No connector definition to match the connector");
+    }
+
+    final AtomicReference<JsonNode> connectorConfigurationJson = new AtomicReference<>(connectorConfiguration.getConfiguration());
+    final List<String> uniqSecretPaths = getSecretPath(connectorConfiguration.getSpec())
+        .stream()
+        .flatMap(secretPath -> getAllExplodedPath(connectorConfigurationJson.get(), secretPath).stream())
+        .toList();
+
+    final UUID workspaceId = connectorConfiguration.getWorkspace();
+    uniqSecretPaths.forEach(secretPath -> {
+      final Optional<JsonNode> secretValue = getValueForPath(connectorConfigurationJson.get(), secretPath);
+      if (secretValue.isEmpty()) {
+        throw new IllegalStateException("Missing secret for the path: " + secretPath);
+      }
+
+      // Only migrate plain text.
+      if (secretValue.get().isTextual()) {
+        final JsonNode stringSecretValue = secretValue.get();
+
+        final SecretCoordinate coordinate =
+            new SecretCoordinate(SecretsHelpers.getCoordinatorBase("airbyte_workspace_", workspaceId, uuidProvider), 1);
```

Comment:
We are still repeating `airbyte_workspace_` in this line and [here](https://github.com/airbytehq/airbyte/blob/aab4f34ca402febcd03cdf0cc81ba9886424edc2/airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretsHelpers.java#L304). Is it possible to change this so that the `airbyte_workspace_` prefix is only defined once?